src/container/internal.tsx

```diff
@@ -49,15 +50,34 @@ export default function InternalContainer({
   const rootRef = useRef<HTMLDivElement>(null);
   const headerRef = useRef<HTMLDivElement>(null);
   const { isSticky, isStuck, stickyStyles } = useStickyHeader(rootRef, headerRef, __stickyHeader, __stickyOffset);
+  const { setHasStickyBackground } = useContext(AppLayoutContext);
   const isRefresh = useVisualRefresh();
+
   const hasDynamicHeight = isRefresh && variant === 'full-page';
   const overlapElement = useDynamicOverlap({ disabled: !hasDynamicHeight });
 
   const mergedRef = useMergeRefs(rootRef, __internalRootRef);
   const headerMergedRef = useMergeRefs(headerRef, overlapElement, __headerRef);
-
   const headerIdProp = __headerId ? { id: __headerId } : {};
 
+  /**
+   * The visual refresh AppLayout component needs to know if a child component
+   * has a high constrast sticky header. This is to make sure the background element
+   * stays in the same vertical position as the header content.
+   */
+  useLayoutEffect(
+    function handleHasStickyBackground() {
+      if (isRefresh && isSticky && variant === 'full-page') {
```

Comment:
Is there a chance that something changes so that we want to change `hasStickyBackground` back from `true` to `false`? For example, an element with sticky header being no longer present.

Reply:
Yes! If the internal container un-mounts due to a route change (or some other reason) that should be handled by the `cleanup` function which resets `hasStickyBackground` back to `false`.

Reply:
Oh I see, this `InternalContainer` is a child of both `Table` and `Card`, so every time the Table or Card unmounts, this will also unmount, do I understand correctly? (I thought it was sort of the other way around, this component being a sort of global container which would eventually contain Tables or Cards)